server/JsDbg.Gdb/GdbDebugger.cs

```diff
@@ -320,7 +321,13 @@ class GdbDebugger : IDebugger {
 
             if (result.Name.StartsWith("vtable for ")) {
                 result.Name = result.Name.Substring("vtable for ".Length) + "::`vftable'";
-                // TODO: If displacement = 2 pointers, assume RTTI and set displacement to 0?
+                ulong pointer_size = IsPointer64Bit ? 8UL : 4UL;
+                if (result.Displacement == 2 * pointer_size) {
+                    // First two words of the vtable are reserved for RTTI.
+                    // http://refspecs.linuxbase.org/cxxabi-1.83.html#rtti-layout
+                    // TODO: Should we subtract 2*pointer_size from all vtable references?
+                    result.Displacement = 0;

```

Comment:
Will this have an adverse impact on other lookups? E.g. if a `bar` parameter is known to be at displacement `16`, is that consistently with/without the RTTI pointers included? If we have a 
```c++
struct TFoo {
  int a;
  int b;
  virtual void baz() {}
}

TFoo foo;
```

then if I `LookupSymbolName(&foo)` does that tell me that I'm at displacement `2*sizeof(void*)`, and is `&(foo->b)` at `2*sizeof(void*) + sizeof(int)`? If we lie and say that the former is at displacement 0, will that mess with trying to find `a` and `b` at some point?

I haven't looked at enough jsdbg code to know exactly how these results are used.

Reply:
It doesn't affect that at all. It's only about where the vtable pointer points to relative to the vtable symbol. For your example:

struct TFoo {
  void* implicit_vtable_pointer;
  int a;
  int b;
};
struct vtable_for_TFoo {
  void* some_rtti_stuff_1;
  void* some_rtti_stuff_2;
  void (*baz)();
};

implicit_vtable_pointer will not be at &vtable_for_TFoo, but will instead point to &vtable_for_TFoo.baz, and only that will be affected.

(The caller in question asks for LookupSymbolName TFoo.implicit_vtable_pointer, not &TFoo)

Reply:
I see, thanks for the clarification!